src/GraphQL/Utilities/AstPrinter.cs

```diff
@@ -319,7 +325,7 @@ public AstPrintVisitor()
                 c.Print(f =>
                 {
                     var val = (decimal)f.Arg(x => x.Value);
-                    return val.ToString("0.0##############", CultureInfo.InvariantCulture);
+                    return val.ToString("G29", CultureInfo.InvariantCulture); //prints most compact form of value without losing any precision
```

Comment:
`decimal` values are designed to have no internal/unseen rounding errors.  So this prints all decimal values with full precision.  Truthfully, this is a bit pointless per spec, as a value with 29 decimal places cannot be stored within an IEEE 754 number.  However, it does not break the spec either.